File: go/simple_example.go
package main

import (
	"bytes"        // Provides functions for manipulating byte slices
	"encoding/json" // Provides JSON encoding and decoding
	"fmt"          // Provides formatting and printing functions
	"net/http"     // Provides HTTP client and server implementations
	"net/url"      // Provides URL parsing and query encoding
)

// main is the entry point of the application
func main() {
	// Define webhook configuration
	webhookID := "YOUR_WEBHOOK_ID"     // Unique identifier for the webhook
	webhookCode := "YOUR_WEBHOOK_CODE" // Authentication code for the webhook

	// Construct the webhook URL
	urlStr := "https://psyll.com/env/webhook/" + webhookID

	// Create form-encoded payload for the webhook request
	data := url.Values{}
	data.Set("code", webhookCode)     // Set webhook authentication code
	data.Set("action", "BUY")         // Specify the action to perform
	data.Set("quantity", "0.0034")    // Set quantity for the transaction
	data.Set("symbol", "BTCUSDT")     // Set trading pair symbol

	// Send POST request with form-encoded data to the webhook URL
	resp, err := http.PostForm(urlStr, data)
	if err != nil {
		// Handle and log errors from the HTTP request
		fmt.Printf("Error: %v\n", err)
		return
	}
	defer resp.Body.Close() // Ensure the response body is closed after use

	// Output the HTTP status code of the response
	fmt.Printf("Status code: %d\n", resp.StatusCode)

	// Decode the JSON response into a map
	var jsonResponse map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&jsonResponse); err != nil {
		// Handle and log errors during JSON decoding
		fmt.Printf("Error decoding JSON: %v\n", err)
		return
	}

	// Marshal the JSON response with indentation for readability
	jsonData, _ := json.MarshalIndent(jsonResponse, "", "    ")
	// Output the formatted JSON response
	fmt.Println("Response:")
	fmt.Println(string(jsonData))
}

File: java/simple_example.java
import java.io.BufferedReader; // Provides buffered reading of character input streams
import java.io.InputStreamReader; // Converts byte streams to character streams
import java.net.HttpURLConnection; // Supports HTTP-specific connection features
import java.net.URL; // Represents a Uniform Resource Locator
import java.net.URLEncoder; // Encodes strings for URL compatibility
import java.nio.charset.StandardCharsets; // Defines standard character encodings
import java.util.HashMap; // Provides a hash table data structure
import java.util.Map; // Interface for key-value pair collections
import java.util.stream.Collectors; // Provides stream operations for collecting results

public class Main {
    // Entry point of the application
    public static void main(String[] args) {
        try {
            // Define webhook configuration
            String webhookId = "YOUR_WEBHOOK_ID"; // Unique identifier for the webhook
            String webhookCode = "YOUR_WEBHOOK_CODE"; // Authentication code for the webhook
            String urlStr = "https://psyll.com/env/webhook/" + webhookId; // Construct the webhook URL

            // Create a map to hold form-encoded payload data
            Map<String, String> payload = new HashMap<>();
            payload.put("code", webhookCode); // Webhook authentication code
            payload.put("action", "BUY"); // Specify the action to perform
            payload.put("quantity", "0.0034"); // Quantity for the transaction
            payload.put("symbol", "BTCUSDT"); // Trading pair symbol

            // Encode payload as URL-encoded form data
            String formData = payload.entrySet().stream()
                .map(e -> {
                    try {
                        // Encode key and value for URL compatibility
                        return URLEncoder.encode(e.getKey(), StandardCharsets.UTF_8.toString()) + "=" +
                               URLEncoder.encode(e.getValue(), StandardCharsets.UTF_8.toString());
                    } catch (Exception ex) {
                        return ""; // Return empty string on encoding error
                    }
                })
                .collect(Collectors.joining("&")); // Join key-value pairs with '&'

            // Create URL object and open HTTP connection
            URL url = new URL(urlStr);
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("POST"); // Set request method to POST
            conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded"); // Set content type
            conn.setDoOutput(true); // Enable output for sending data

            // Write form data to the request body
            try (var outputStream = conn.getOutputStream()) {
                outputStream.write(formData.getBytes(StandardCharsets.UTF_8));
            }

            // Retrieve and print the HTTP status code
            int statusCode = conn.getResponseCode();
            System.out.println("Status code: " + statusCode);

            // Read and build the response from the server
            StringBuilder response = new StringBuilder();
            try (BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()))) {
                String line;
                // Read each line of the response until none remain
                while ((line = reader.readLine()) != null) {
                    response.append(line);
                }
            }

            // Output the response content
            System.out.println("Response:");
            System.out.println(response.toString());

            // Close the connection
            conn.disconnect();
        } catch (Exception e) {
            // Handle and log any exceptions that occur during execution
            System.err.println("Error: " + e.getMessage());
        }
    }
}